src/routes/bulk/post.rs

```diff
@@ -213,14 +273,9 @@ async fn create_bulk_request(
 	})?;
 
 	for task_input in body.into_iter() {
-		let task = TaskInput {
-			input: task_input,
-			job_id: rec.id,
-		};
-
 		let task_uuid = email_verification_task
 			.builder()
-			.set_json(&task)
+			.set_json(&(rec.id, task_input))
 			.unwrap()
```

Comment:
There's no good way to extract a value from a Result without panicing. `map_err` or `expect` only helps in changing the error type or the panic message, there will still be a panic.

The only other way is to do an `if let` or `match case` and swallow the error case. But that means putting everything inside the if/case indented scope.

Usually it's fine to have unwraps/expects in the code because they do mean that that case should never occur. This is especially true in async programs because the whole problem does not abort only the particular thread gets terminated and the recovers.

Reply:
I think a map_err in order to log the error message is the best way to go ahead. What do you think.

Reply:
Can't we `.set_json(&(rec.id, task_input))?` here (with potentially a map_err)?

If not, then let's map_err and log as you suggest.

Reply:
Fixed.